tanjun/checks.py

```diff
@@ -787,3 +788,63 @@ def with_check(check: tanjun_abc.CheckSig, /) -> collections.Callable[[CommandT]
         A command decorator callback which adds the check.
     """
     return lambda command: command.add_check(check)
+
+
+def with_any_role_check(
+    roles: list[str | int] = list(),
+    *,
+    error_message: str | None = "You do not have the required roles to use this command!",
+    halt_execution: bool = True,
+) -> collections.Callable[[CommandT], CommandT]:
+    """Only let a command run if the author has a specific role.
+
+    Parameters
+    ----------
+    roles: list[str | int]
+        The author must have at least one (1) role in this list. (Role.name and Role.id are checked)
```

Comment:
rather than `int` can we instead accept `hikari.SnowflakeishOr[hikari.Role]` (this resolves to [int | hikari.Role | Snowflake]) and then cast entries in the list which aren't instance of str to hikari.Snowflake, this'd keep it more inline with how places where we only really need the resource's ID are generally handled between Tanjun and Hikari.